presto-main/src/main/java/com/facebook/presto/execution/CreateFunctionTask.java

```diff
@@ -80,7 +81,11 @@ private SqlInvokedFunction createSqlInvokedFunction(CreateFunction statement)
             throw new PrestoException(GENERIC_USER_ERROR, format("Invalid function name: %s, require exactly 3 parts", statement.getFunctionName()));
         }
 
-        QualifiedFunctionName functionName = QualifiedFunctionName.of(statement.getFunctionName().toString());
+        QualifiedFunctionName functionName = QualifiedFunctionName.of(
+                new CatalogSchemaName(
+                        statement.getFunctionName().getParts().get(0),
```

Comment:
nit: for readability, introduce a local variable for `statement.getFunctionName().getParts()` and use it here and in the check above